File: Ejercicio9.hs
module Ejercicio9 where

import System.IO



pedidoPizza :: String -> String -> (String, Bool)
pedidoPizza eleccion ingrediente
    | eleccion == "vegetariana" = ("Pizza vegetariana con " ++ ingrediente, True)
    | otherwise                 = ("Pizza no vegetariana con " ++ ingrediente, False)

main :: IO ()
main = do
    hSetBuffering stdout NoBuffering
    putStr "¿Quiere una pizza vegetariana? (si/no): "
    eleccion <- getLine
    putStr "Ingrese un ingrediente adicional: "
    ingrediente <- getLine
    let (descripcion, esVegetariana) = pedidoPizza eleccion ingrediente
    putStrLn $ "Descripción de la pizza: " ++ descripcion
    putStrLn $ "¿Es vegetariana?: " ++ show esVegetariana

File: Ejercicio9.py

def pedido_pizza():
    eleccion_vegetariana = input("¿Quieres una pizza vegetariana? (si/no): ").lower()

    if eleccion_vegetariana == "si":
        print("\nIngredientes vegetarianos disponibles: champiñones, pimientos, cebolla, aceitunas.")
        ingrediente_adicional = input("Elige un ingrediente adicional vegetariano: ")
        descripcion_pizza = f"Pizza vegetariana con {ingrediente_adicional}"
        es_vegetariana = True
    else:
        print("\nIngredientes no vegetarianos disponibles: pepperoni, jamón, pollo, carne.")
        ingrediente_adicional = input("Elige un ingrediente adicional no vegetariano: ")
        descripcion_pizza = f"Pizza no vegetariana con {ingrediente_adicional}"
        es_vegetariana = False

    print(f"\nDescripción final de la pizza: {descripcion_pizza}")
    print(f"¿Es vegetariana?: {es_vegetariana}")

if __name__ == "__main__":
    pedido_pizza()


